setup.py

```diff
@@ -1,12 +1,146 @@
-from setuptools import find_packages, setup
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
 
+"""
+Module that defines a setup function and publishes the package to PyPI.
+
+Use the command `python setup.py upload`.
+"""
+# Note: To use the "upload" functionality of this file, you must:
+#   $ pip install twine
+
+import io
+import os
+import sys
+from shutil import rmtree
+from typing import Dict, List
+
+from setuptools import Command, find_packages, setup
+
+# Package meta-data.
+NAME = "geograph"
+DESCRIPTION = "Group Team Challenge 2021 - Biodiversity Team"
+URL = "https://geograph.readthedocs.io/"
+EMAIL = "hb574@cam.ac.uk"
+AUTHOR = "Biodiversity Team"
+REQUIRES_PYTHON = ">=3.8.0"
+
+# What packages are required for this module to be executed?
+REQUIRED: List = [
+    "numpy",
+    "pandas",
+    "folium",
+    "ipyleaflet",
+    "tqdm",
+    "geopandas",
+    "shapely",
+    "rtree",
+    "rasterio",
+    "xarray",
+    "networkx",
+]
```

Comment:
The versions of these dependencies are not pinned at the moment right? That's probably something to do for future releases -> v.0.0.2 and onwards...

Reply:
Yes, but it is usually not considered best practice to pin exact dependencies in `setup.py`, since it is intended as a list of the minimal requirements that are to be as flexible as possible: https://stackoverflow.com/a/44938662. Pinning dependencies to ranges would be fine of course, but at the moment we don't know what the ranges should be.